apps/src/netsim/NetSimLocalClientNode.js

```diff
@@ -457,7 +457,13 @@ NetSimLocalClientNode.prototype.sendMessages = function (payloads, onComplete) {
  * @private
  */
 NetSimLocalClientNode.prototype.onNodeTableChange_ = function (nodeRows) {
-  // 1. Remove simulating routers that have vanished from remote storage.
+  // If our own row is gone, drop everything and handle disconnect.
+  if (this.onNodeLostConnection_ && !this.canFindOwnRowIn(nodeRows)) {
+    this.onNodeLostConnection_();
+    return;
+  }
```

Comment:
I thought about this while editing, myself.  The truth is, we should always have an onNodeLostConnection_ function in production so this check might be overkill.  We could change it to be an empty function by default so it's always valid.  There's a redundant set check, but it's probably unneeded.